### ISSUE TITLE
Confusing behaviour of `is_vector`

### ISSUE THREAD

**Issue by olivierverdier** [JuliaManifolds/Manifolds.jl#613]
The behaviour of `is_vector` may be extremely confusing to newcomers. It seems that after a very superficial check, it will return `true` by default?

```julia
using Manifolds

G = SpecialOrthogonal(3) # any manifold will do, here

@show is_point(G, Identity(G)) # true: good!
@show is_point(G, 2*Identity(G)) # false: good!

@show is_vector(G, Identity(G), randn(3,3)) # true ?!
@show is_vector(G, Identity(G), [2,3]) # true ?!
@show is_vector(G, Identity(G), "hello") # true ?!
```

I'm sure I'm doing something wrong, but I hope you can see how confusing this may be.

It may even be a feature? In `ManifoldsBase`, the doc and implementation of `check_vector` is

```julia
"""
By default, `check_vector` returns `nothing`, i.e. if no checks are implemented, the
assumption is to be optimistic for tangent vectors not deriving from the [`TVector`](@ref)
type.
"""
check_vector(M::AbstractManifold, p, X; kwargs...) = nothing
```

Solutions:
1. Change the name to `possible_vector`, `potential_vector`, etc., but not `is_vector`
2. Remove the `nothing` implementation of `check_vector`? I believe most users would prefer a loud fail (like, missing method `check_vector` for the given type) to a silent fail anytime?

**Comment by kellertuer** [comment]
Hi,

that is indeed a design choice, since `check_vector` (the internal function) returns `nothing` by default. The reason is, that for example on `Euclidean(n)` there is nothing to check (besides the size, but that is a separate check). So that default means “If the size is ok, which I do not check – then everything is ok” which is our default assumption.

Your three cases that should fail, do not do that since `check_size` (which could be called `check_size_and_type` but that sounds a bit clumsy. Maybe `check_size` is not good-enough implemented here.
The reason for this split is, that `check_size` is something `is_point` _and_ `is_vector` have in common to check (see https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/0225e499b7569c40d2a45aa762bc7c5ed6c499f8/src/ManifoldsBase.jl#L664 and https://github.com/JuliaManifolds/ManifoldsBase.jl/blob/0225e499b7569c40d2a45aa762bc7c5ed6c499f8/src/ManifoldsBase.jl#L728.

I can take a look later today where we are missing `check_size` checks for your counter examples for SO(n).

Concerning
1. I am not a fan of your function names proposed, since they would be very inprecise.
2. No, since `check_vector` is not the only check done, it just says “besides the size there is nothing to check” - or in other words “If the size is right, then all is fine” by default, hence its `nothing`-default

**Comment by kellertuer** [comment]
First small update, it is the dispatch with the `identity(G)`, where we use a type instead of (here) the unit matrix to space some space, that is

```Julia
julia> is_vector(G, identity_element(G), randn(3,3))
false

julia> is_vector(G, identity_element(G), [2,3])
false

julia> is_vector(G, identity_element(G), "hello")
ERROR: MethodError: no method matching size(::String)
```

as soon as we use the identity matrix. So this is mainly a but somewhere in the dispatch. Will narrow that down later, probably one just has to ”materialize“ the identity element to its matrix form as I did here manually.

**Comment by olivierverdier** [comment]
Well done! Thank you for fixing this so quickly!

**Comment by kellertuer** [comment]
Was a good escape from grading twice this morning ;)
Thanks for spotting that small mistake (and interesting that none of our tests hit that); but since I know the structure of the packages a bit it was relatively easy to “hunt down”.

**Comment by kellertuer** [comment]
The fix is now published in 0.8.65
